pysb/simulator/scipyode.py

```diff
@@ -403,36 +403,42 @@ def _test_inline(cls):
                                      extra_compile_args=extra_compile_args)
                     cls._use_inline = True
                 except (weave.build_tools.CompileError,
-                        distutils.errors.CompileError, ImportError):
-                    pass
-                except ValueError as e:
-                    if len(e.args) == 1 and \
-                                        e.args[0] == "Symbol table not found":
-                        get_logger(__name__).debug(
-                            "'ValueError: Symbol table not found' "
-                            "encountered; weave compiler is not functional")
-                    else:
+                        distutils.errors.CompileError,
+                        ImportError,
+                        ValueError) as e:
+                    if isinstance(e, ValueError) and (
```

Comment:
This logic was already a little obscure and it's duplicated here and below. The immediate debug message previously made it fairly clear what was being checked, but with this change it's less apparent what's going on. Maybe this is a good time to refactor it. Could you refactor it into a well-named predicate function on `e` with an explanation of what it's looking for? Also I don't think rewriting the arg-checking term as a disjunction of negations made it any better... I do prefer the old form. Or maybe just `e.args == ('Symbol table not found',)` which actually covers the whole thing at once?